### ISSUE TITLE
FP `unused_import` when the import is only used in rustdoc links

### ISSUE THREAD

**Issue by jendrikw** [rust-lang/rust#92493]
I tried this code:

```rust
//! This is a link: [`Add::add`].
use std::ops::Add;
```

I expected to see this happen: No warning

Instead, this happened:
```
warning: unused import: `std::ops::Add
```

This is especially annoying when you deny some lints:

`cargo build` errors, but `cargo doc` is fine:

```rust
#![deny(unused_imports)]
#![deny(rustdoc::broken_intra_doc_links)]
//! This is a link: [`Add::add`].
use std::ops::Add;
```

When you comment out the use statement, `cargo build` is fine, but `cargo doc` errors.
```rust
#![deny(unused_imports)]
#![deny(rustdoc::broken_intra_doc_links)]
//! This is a link: [`Add::add`].
//use std::ops::Add;
```

I know I can use the links syntax with an explicit target, but I prefer the short form for less noise when reading the code.

`rustc --version --verbose`:
```
rustc 1.57.0 (f1edd0429 2021-11-29)
binary: rustc
commit-hash: f1edd0429582dd29cccacaf50fd134b05593bd9c
commit-date: 2021-11-29
host: x86_64-unknown-linux-gnu
release: 1.57.0
LLVM version: 13.0.0
```

**Comment by jendrikw** [comment]
Oops, dupe of #79542.
